tidb_vector/integrations/vector_client.py

```diff
@@ -296,26 +299,62 @@ def _vector_search(
         query_embedding: List[float],
         k: int = 5,
         filter: Optional[Dict[str, str]] = None,
+        **kwargs: Any,
     ) -> List[Any]:
         """vector search from table."""
 
-        filter_by = self._build_filter_clause(filter)
+        post_filter_enabled = kwargs.get("post_filter_enabled", False)
+        post_filter_multiplier = kwargs.get("post_filter_multiplier", 1)
         with Session(self._bind) as session:
-            results: List[Any] = (
-                session.query(
-                    self._table_model.id,
-                    self._table_model.meta,
-                    self._table_model.document,
-                    self.distance_strategy(query_embedding).label("distance"),
+            if post_filter_enabled is False or filter is None:
```

Comment:
```suggestion
            if post_filter_enabled is False or not filter:
```
User may pass in either empty dict or `None`